code/modules/cargo/expressconsole.dm

```diff
@@ -1,41 +1,63 @@
 #define MAX_EMAG_ROCKETS 8
+#define BEACON_COST 5000
+#define SP_LINKED 1
+#define SP_READY 2
+#define SP_LAUNCH 3
+#define SP_UNLINK 4
+#define SP_UNREADY 5
 
 /obj/machinery/computer/cargo/express
 	name = "express supply console"
 	desc = "This console allows the user to purchase a package \
-		with 1/40th of the delivery time: made possible by NanoTrasen's new \"Drop Pod Railgun\".\
+		with 1/40th of the delivery time: made possible by NanoTrasen's new \"1500mm Orbital Railgun\".\
 		All sales are near instantaneous - please choose carefully"
 	icon_screen = "supply_express"
 	circuit = /obj/item/circuitboard/computer/cargo/express
 	blockade_warning = "Bluespace instability detected. Delivery impossible."
 	req_access = list(ACCESS_QM)
 	var/message
-	var/locked = TRUE
+	var/printed_beacons = 0 //number of beacons printed. Used to determine beacon names.
 	var/list/meme_pack_data
-	var/podID = 0//0 is your standard supply droppod (requires dissassembly after landing), 1 is the bluespace drop pod (teleports out after landing)
+	var/obj/item/supplypod_beacon/beacon //the linked supplypod beacon
+	var/area/landingzone = /area/quartermaster/storage //where we droppin boys
+	var/podID = POD_STANDARD //0 is your standard supply supplypod (requires dissassembly after landing), 1 is the bluespace supply pod (teleports out after landing)
+	var/cooldown = 0 //cooldown to prevent printing supplypod beacon spam
+	var/locked = TRUE //is the console locked? unlock with ID
+	var/usingBeacon = FALSE //is the console in beacon mode? exists to let beacon know when a pod may come in
 
 /obj/machinery/computer/cargo/express/Initialize()
 	. = ..()
 	packin_up()
 
+/obj/machinery/computer/cargo/express/Destroy()
+	if(beacon)
+		beacon.unlink_console()
+	return ..()
 
 /obj/machinery/computer/cargo/express/attackby(obj/item/W, mob/living/user, params)
 	if((istype(W, /obj/item/card/id) || istype(W, /obj/item/pda)) && allowed(user))
 		locked = !locked
 		to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the interface.</span>")
 		return
 	else if(istype(W, /obj/item/disk/cargo/bluespace_pod))
-		podID = 1//doesnt effect circuit board, so that reversal is possible
+		podID = POD_BLUESPACE//doesnt effect circuit board, making reversal possible
 		to_chat(user, "<span class='notice'>You insert the disk into [src], allowing for advanced supply delivery vehicles.</span>")
 		qdel(W)
 		return TRUE
+	else if(istype(W, /obj/item/supplypod_beacon))
+		var/obj/item/supplypod_beacon/sb = W
+		if (sb.express_console != src)
+			sb.link_console(src, user)
+			return TRUE

```

Comment:
Should move this out of the `if` so already being linked doesn't make you hit the console.

Reply:
that was intentional: players would recieve a response when unsuccessfully linking the device

Reply:
It's fine to get a chat message but a failed link shouldn't also attack the console.